### DISCUSSION TITLE
Reading SCIP solutions

### DISCUSSION THREAD

**Discussion by th-yoon** [ds4dm/ecole#208] [Questions] [answered]
Hi, I'm trying to read solutions written from SCIP into ecole framework.
Do you guys support this feature?

**Comment by AntoinePrv** [comment, marked as answer]
Hi @th-yoon,

You can access the `Model` of the environment and convert it to a PySCIPOpt one to query all information you want.
```python
env.model.as_pyscipopt()
```

You can also make it into an [`information_function`](https://doc.ecole.ai/py/en/stable/howto/create-functions.html) that you pass to the environment if you wish to compute it systematically.

**Reply by th-yoon** [reply]
Thank you for answering my question. I knew that scip has a function that reads the solution file, but I wondered if there's a built in preprocessor in the ecole framework. It seems I need to preprocess it anyways :smile:

**Reply by gasse** [reply]
Hello @th-yoon,

No that's not a functionality we provide at the moment. Our philosophy is that if what you want to achieve can be done easily by using Ecole + PySCIPOpt, then there is no need to duplicate PySCIPOpt's features into Ecole. If what you want to do ends up being non-trivial with the PySCIPOpt API, and you think we should include it into Ecole, let us know.

Best,
Maxime

**Reply by th-yoon** [reply]
Hi @gasse,
I understand. Thank you for your support!
